app/addons/setup/setup.js

```diff
@@ -22,11 +22,41 @@ var ConfirmButton = ReactComponents.ConfirmButton;
 
 
 var ClusterConfiguredScreen = React.createClass({
+  getInitialState: function () {
+    return this.getStoreState();
+  },
+
+  getStoreState: function () {
+    return {
+      clusterState: setupStore.getClusterState()
+    };
+  },
+
+  componentDidMount: function () {
+    setupStore.on('change', this.onChange, this);
+  },
+
+  componentWillUnmount: function () {
+    setupStore.off('change', this.onChange);
+  },
+
+  onChange: function () {
+    this.setState(this.getStoreState());
+  },
 
   render: function () {
+    var nodetype;
+
+    if (this.state.clusterState === 'cluster_finished') {
+      nodetype = 'clustered';
+    } else if (this.state.clusterState === 'single_node_enabled') {
+      nodetype = 'single';
+    } else {
+      nodetype = 'unknown state';
```

Comment:
you can put this if-else in a function to clean up the render function.

like 

var nodetype = this.getNodeType();